ruby/lib/google/protobuf/repeated_field.rb

```diff
@@ -29,11 +29,141 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 # add syntatic sugar on top of the core library
+require 'forwardable'
 module Google
   module Protobuf
     class RepeatedField
+      extend Forwardable
 
-      alias_method :size, :length
+      # methods defined in C or Java:
+      #   +
+      #   [], at
+      #   []=
+      #   clear
+      #   dup, clone
+      #   each
+      #   each_with_index
+      #   hash
+      #   insert
+      #   push, <<
+      #   pop
+      #   replace
+      #   length, size
+      #   ==
+      #   to_ary, to_a
+      #   also all enumerable
+
+      # NOTE:  using delegators rather than method_missing to make the
+      #        relationship explicit instead of implicit
+      # NOTE2: use to_a instea
+      def_delegators :to_ary,
+        :&, :*, :-, :'<=>',
+        :assoc, :bsearch, :combination, :compact, :count, :cycle,
+        :drop, :drop_while, :eql?, :fetch, :find_index, :flatten,
+        :include?, :index, :inspect, :join,
+        :pack, :permutation, :product, :pretty_print, :pretty_print_cycle,
+        :rassoc, :repeated_combination, :repeated_permutation, :reverse, :rindex, :rotate,
+        :sample, :shelljoin, :slice, :to_s, :transpose, :uniq, :|
```

Comment:
the other way to do this is with a `method_missing` and just pass any unmapped call to `to_ary`.  However, this will lead to unexpected behavior if we missed, or ruby 3.0 adds a new method that modifies self.  So I decided to be explicit